### DISCUSSION TITLE
How to install properly ImageMagick 7 in Ubuntu to have all needed delegates?

### DISCUSSION THREAD

**Discussion by gerkas** [ImageMagick/ImageMagick#6981] [Help] [answered]
### ImageMagick version

7.1.1-9

### Operating system, version and so on

Linux, Ubuntu 22.04 LTS

### Description

Hi all,

I'm trying to install IM 7 in Ubuntu 22.04 and in Cygwin. I´ve tried with easiest way with

`sudo apt install ImageMagick`

But I note that FFTW delegate is not installed nor reconized even I installed previosly FFTW3 with these commands

`sudo apt-get update -y
sudo apt-get install -y fftw3` 

Then I uninstalled and installed again from source the version 7.1.1-19 and FFTW doesn´t appear in Delegates.

How to fix this? Thank you

`Version: ImageMagick 7.1.1-19 Q16-HDRI x86_64 21601 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP(4.5)  
Delegates (built-in): bzlib djvu fontconfig freetype gvc jbig jng jp2 jpeg lcms lqr lzma openexr pangocairo png raqm raw tiff webp x xml zip zlib
Compiler: gcc (11.4)`

**Comment by fmw42** [comment]
Does this help?  https://gist.github.com/makenova/78bb63aaa1050e2ad8019ee1e7e7b433

**Reply by gerkas** [reply]
Thanks for the link. I uninstalled and begin to install again from source following the steps in link you shared. I go in `./configure` step and FFTW appears in NO in Delegate library configuration: How to put in yes?

Delegate library configuration:
  BZLIB             --with-bzlib=yes                    yes
  Autotrace         --with-autotrace=no                 no
  DJVU              --with-djvu=yes                     yes
  DPS               --with-dps=no                       no
  FFTW              --with-fftw=no                      no
  FLIF              --with-flif=no                      no
  FlashPIX          --with-fpx=no                       no
  FontConfig        --with-fontconfig=yes               yes

**Reply by fmw42** [reply]
I am not expert on Linux, but you must install FFTW delegate library first (where Imagemagick can find it), I suppose.

**Reply by gerkas** [reply]
I already installed it. Not sure why is not being recognized

`$ sudo apt-get install libfftw3-dev
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
libfftw3-dev is already the newest version (3.3.8-2ubuntu8).
0 upgraded, 0 newly installed, 0 to remove and 2 not upgraded.`

I saw [this thread](https://legacy.imagemagick.org/discourse-server/viewtopic.php?t=31205) in IM forum and for that guy was fixed after installing FFTW3, not for me.

**Comment by snibgo** [comment]
> FFTW --with-fftw=no no

The first "no" means you didn't ask for it. The second "no" means you didn't get it.

To ask for it, include a `./configure` option `--with-fftw=yes`.

If you ask for it and don't get it, check `config.log` for the reason.

**Comment by gerkas** [comment, marked as answer]
Thank you for the suggestion. I checked the `config.log` and mentioned something like `Package FFTW3 was not found in the pkg-config search path`  then I installed these packages 

`sudo apt-get install libfftw3-dev`
`sudo apt-get install libsndfile1-dev`

Following the suggestion in [this thread](https://askubuntu.com/questions/680108/package-was-not-found-in-the-pkg-config-search-path) and I sent `./configure --with-fftw=yes` again and FFTW was recognized.
